src/pykka/_actor.py

```diff
@@ -10,6 +12,30 @@
 logger = logging.getLogger("pykka")
 
 
+class TimedInbox:
+    def __init__(self):
+        self.timestamps = []
+        self.envelopes = []
+
+    def empty(self):
+        return len(self.timestamps) == 0
+
+    def add(self, envelope):
+        idx = bisect.bisect(self.timestamps, envelope.timestamp)
+        self.timestamps.insert(idx, envelope.timestamp)
+        self.envelopes.insert(idx, envelope)
+
+    def pop(self):
+        self.timestamps.pop(0)
+        return self.envelopes.pop(0)
+
+    def next_event_in(self):
+        if len(self.timestamps) == 0:
+            return None
+        else:
+            return max(self.timestamps[0] - time.monotonic(), 0)
```

Comment:
Not sure what kind of API is needed for that, given that it has to match the call in `_envelope.py`. Perhaps it will be easier to just mock `time.monotonic` in tests?

Reply:
You might want to read up on the clock pattern:
https://agilewarrior.wordpress.com/2017/03/03/clock-pattern/

Reply:
So how is that different from mocking `time.monotonic`?

Reply:
Mocking time.monotonic() in particular may not be feasible. A dedicated clock object can provide more flexibility in this regard, as well as a nicer interface to manage the clock in tests.

Reply:
I guess that's true. But there's still a problem of similarly affecting the timeouted methods of `Queue` or `Event`.